src/btree/bt_sync.c

```diff
@@ -143,6 +143,42 @@ __sync_ref_mark_deleted(WT_SESSION_IMPL *session, WT_REF *ref)
     return (0);
 }
 
+/*
+ * __sync_ref_int_obsolete_cleanup --
+ *     Traverse the internal page and identify the leaf pages that are obsolete and mark them as
+ *     deleted.
+ */
+static int
+__sync_ref_int_obsolete_cleanup(WT_SESSION_IMPL *session, WT_REF *intref)
+{
+    WT_PAGE_INDEX *pindex;
+    WT_REF *ref;
+    uint32_t slot;
+
+    pindex = NULL;
+
+    pindex = WT_INTL_INDEX_GET_SAFE(intref->page);
+    for (slot = 0; slot < pindex->entries; slot++) {
+        /*
+         * Get a reference, setting the reference hint if it's wrong (used when we continue the
+         * walk). We don't always update the hints when splitting, it's expected for them to be
+         * incorrect in some workloads.
+         */
+        ref = pindex->index[slot];
+        if (ref->pindex_hint != slot)
+            ref->pindex_hint = slot;
```

Comment:
I don't understand why we're resetting the reference hints in this loop? I mean, it's not wrong, but I'm not sure why we want to do that work here?

Reply:
Already the tree walk function has taken care of resetting the slot that are not matching, this code is not required. Removed.